src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -5394,6 +5417,253 @@ def delete_deployment(self, deployment_id: UUID) -> None:
+                resource_project_id = resource_schema.project_id
+                if resource_project_id and resource_project_id != project_id:
+                    raise IllegalOperationError(
+                        f"Resource {visualization.resource_type.value} with ID "
+                        f"{visualization.resource_id} belongs to a different project than "
+                        f"the curated visualization (project ID: {project_id})."
+                    )
+
+            self._assert_curated_visualization_duplicate(
+                session=session,
+                artifact_version_id=visualization.artifact_version_id,
+                visualization_index=visualization.visualization_index,
+                resource_id=visualization.resource_id,
+                resource_type=visualization.resource_type,
+            )
+
+            schema: CuratedVisualizationSchema = (
+                CuratedVisualizationSchema.from_request(visualization)
+            )
+            schema.project_id = project_id
+
+            session.add(schema)
+            session.commit()
+            session.refresh(schema)
+
+            return schema.to_model(
+                include_metadata=True,
+                include_resources=True,
+            )
+
+    def get_curated_visualization(
+        self, visualization_id: UUID, hydrate: bool = True
+    ) -> CuratedVisualizationResponse:
+        """Fetch a curated visualization by ID.
+
+        Args:
+            visualization_id: The ID of the curated visualization to fetch.
+            hydrate: Flag deciding whether to hydrate the output model(s)
+                by including metadata fields in the response.
+
+        Returns:
+            The curated visualization with the given ID.
+        """
+        with Session(self.engine) as session:
+            schema: CuratedVisualizationSchema = self._get_schema_by_id(
+                resource_id=visualization_id,
+                schema_class=CuratedVisualizationSchema,
+                session=session,
+            )
+            return schema.to_model(
+                include_metadata=hydrate,
+                include_resources=hydrate,
+            )
+
+    def update_curated_visualization(
+        self,
+        visualization_id: UUID,
+        visualization_update: CuratedVisualizationUpdate,
+    ) -> CuratedVisualizationResponse:
+        """Update mutable fields on a curated visualization.
+
+        Args:
+            visualization_id: The ID of the curated visualization to update.
+            visualization_update: The update to apply to the curated visualization.
+
+        Returns:
+            The updated curated visualization.
+        """
+        with Session(self.engine) as session:
+            schema = self._get_schema_by_id(
+                resource_id=visualization_id,
+                schema_class=CuratedVisualizationSchema,
+                session=session,
+            )
+            schema.update(visualization_update)
```

Comment:
What happens if I update a curated visualization to have the same display index as another one? Does that cause issues?

Reply:
maybe question for @Cahllagerfeld? how are we expecting to send this from front-end